src/seedu/addressbook/Main.java

```diff
@@ -79,26 +88,62 @@ private void exit() {
     }
 
     /** Reads the user command and executes it, until the user issues the exit command.  */
-    private void runCommandLoopUntilExitCommand() {
+    private void runCommandLoopUntilExitCommand() throws UniquePersonList.DuplicatePersonException {
         Command command;
+        CommandResult result;
+        deletes = new PreviousDeletes();
         do {
             String userCommandText = ui.getUserCommand();
             command = new Parser().parseCommand(userCommandText);
-            CommandResult result = executeCommand(command);
+            result = getCommandResult(command, userCommandText);
             recordResult(result);
             ui.showResultToUser(result);
 
         } while (!ExitCommand.isExit(command));
     }
 
+    /**
+     *
+     * @param command
+     * @param userCommandText
+     * @return result of Command
+     */
+
+    private CommandResult getCommandResult(Command command, String userCommandText) {
+        CommandResult result;
+        if(userCommandText.contains("delete")||userCommandText.contains("undoDelete")) {
+            result = executeCommand(command,deletes);
+        }
+        else {
```

Comment:
Coding standard violation here. Please check our coding standards on how to write conditionals